Source/ARTRealtime.m

```diff
@@ -550,12 +550,6 @@ - (ARTEventListener *)transitionSideEffects:(ARTConnectionStateChange *)stateCha
 
     if ([self shouldSendEvents]) {
         [self sendQueuedMessages];
-        // For every Channel
-        for (ARTRealtimeChannel* channel in self.channels.nosyncIterable) {
-            if (channel.state_nosync == ARTRealtimeChannelSuspended) {
-                [channel _attach:nil];
-            }
```

Comment:
I did the fix in this PR because the RTP17b test was failing because of this. Discussion [here](https://ably-real-time.slack.com/archives/C030C5YLY/p1550849880011900).

Reply:
@ricardopereira With this removal, does the library comply with RTN15c3? "when a resume request results in a CONNECTED with a new connectionId ... The client library should initiate an attach for channels that are in the SUSPENDED state. For all channels in the ATTACHING or ATTACHED state, the client library should fail any previously queued messages for that channel and initiate a new attach..."

cf my comment in slack "you shouldn't be sending an attached after a successful resume, only for an unsuccessful one" -- you still need to do the latter.

(From that test log it looks like the connection moved to `suspended`, but it's not obvious why from the log, the resume was successful so I can't see any reason for it to do so)

Reply:
> With this removal, does the library comply with RTN15c3?

@SimonWoolf Yes it does. When I implemented `RTN15c3`, I forgot to remove that bit because the new implementation already handle that case. You can see that the channel will reattach https://github.com/ably/ably-cocoa/blob/8dbc03d9e1ec9776e4fbeb58a2d85469edc9182f/Source/ARTRealtime.m#L608-L615 and it does check the SUSPENDED channel state https://github.com/ably/ably-cocoa/blob/8dbc03d9e1ec9776e4fbeb58a2d85469edc9182f/Source/ARTRealtimeChannel.m#L924-L934

Reply:
> (From that test log it looks like the connection moved to suspended, but it's not obvious why from the log, the resume was successful so I can't see any reason for it to do so)

@SimonWoolf That's because the test forces to suspend:

https://github.com/ably/ably-cocoa/blob/8dbc03d9e1ec9776e4fbeb58a2d85469edc9182f/Spec/RealtimeClientPresence.swift#L3034-L3039

Reply:
> I forgot to remove that bit because the new implementation already handle that case

:+1:

Though looking at `reattachWithReason`, it looks to me like it's reattaching channels in every state _except_ attaching. So (1) it's starting an attach for channels in `initialized`, `detached`, or `failed`, which it really shouldn't be; and (2) it isn't initiating a new attach for a channel in the `attaching` state (we've just started a completely new connection, so any previously-in-progress attach needs to be started again). Or am I misreading it?

Reply:
@SimonWoolf You're not misreading. It seems so, it's not respecting `RTN15c3`. Issue has been created: #847